integration/setup/setup.go

```diff
@@ -0,0 +1,111 @@
+// +build k8srequired
+
+package setup
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/giantswarm/e2e-harness/pkg/framework"
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/azure-operator/client"
+	"github.com/giantswarm/azure-operator/integration/teardown"
+	"github.com/giantswarm/azure-operator/integration/template"
+)
+
+const (
+	azureResourceValuesFile = "/tmp/azure-operator-values.yaml"
+)
+
+// WrapTestMain setup and teardown e2e testing environment.
+func WrapTestMain(c *client.AzureClientSet, g *framework.Guest, h *framework.Host, m *testing.M) {
+	var r int
+
+	err := Setup(c, g, h)
+	if err != nil {
+		log.Printf("%#v\n", err)
+		r = 1
+	} else {
+		r = m.Run()
+	}
+
+	if os.Getenv("KEEP_RESOURCES") != "true" {
+		teardown.Teardown(c, g, h)
+	}
+
+	os.Exit(r)
+}
+
+// Setup e2e testing environment.
+func Setup(c *client.AzureClientSet, g *framework.Guest, h *framework.Host, m *testing.M) error {
+	var v int
+	var err error
+
+	err = h.Setup()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = Resources(c, g, h)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = g.Setup()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+// Resources install required charts.
+func Resources(c *client.AzureClientSet, g *framework.Guest, h *framework.Host) error {
+	var err error
+
+	{
+		err = h.InstallStableOperator("cert-operator", "certconfig", template.CertOperatorChartValues)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		err = h.InstallCertResource()
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	{
+		err = h.InstallBranchOperator("azure-operator", "azureconfig", template.AzureOperatorChartValues)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		err = installAzureResource()
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	return nil
+}
+
+func installAzureResource() error {
+	azureResourceChartValuesEnv := os.ExpandEnv(template.AzureResourceChartValues)
+	d := []byte(azureResourceChartValuesEnv)
+
+	err := ioutil.WriteFile(azureResourceValuesFile, d, 0644)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = framework.HelmCmd("registry install quay.io/giantswarm/azure-resource-chart -- -n azure-resource-lab --values " + azureResourceValuesFile)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
```

Comment:
Shouldn't you wait here for the resources to be available? It can be racy if not.

Reply:
Yes, sorry for not being more clear, if this is the main CR then the operator itself will wait for it to be ready. I think we should have a generic way to install resources in the framework, and some of them need to wait for some condition to be true in order to prevent races, see https://github.com/giantswarm/e2e-harness/blob/master/pkg/framework/host.go#L162-L175. Not related to this PR anyway.